src/diskio/immediate.rs

```diff
@@ -181,30 +181,30 @@ impl IncrementalFileWriter {
         if (self.state.lock().unwrap()).is_none() {
             return false;
         }
-        let chunk = if let FileBuffer::Immediate(v) = chunk {
```

Comment:
The `else` drop is significant here, however the `else` branch only has an `unreachable!()`.